doc/design/to_run_stmt.md

```diff
@@ -0,0 +1,178 @@
+  
+- Distributed cluster execution  
+  The PyODPS node only installed Mars package, but not Dask and Ray. So we can create a cluster, build the computing graph and submit the distributed execution only using Mars API inside `a_python_func`.
+
+### How to build TO RUN function
+
+#### Kubernetes
+
+```TXT
+-- my_func
+---- data_preprocessor.py
+---- utils.py (optional)
+-- Dockerfile
+```
+
+`a_python_func` is implemented in `data_preprocessor.py`. `docker build -f Dockerfile .` will install the dependency and copy `my_func` folder into `/run/my_func` in the image. `utils.py` contains some helper utils to be used in `data_preprocessor.py`.
+
+#### MaxCompute
+
+```TXT
+-- my_func
+---- data_preprocessor.py
+---- utils.py (optional)
+-- Dockerfile
+```
+
+Build a computing graph using Mars API and submit into mars cluster. `data_preprocessor.py` contains only one function `a_python_func`.
+
+```Python
+def a_python_func(param_a, param_b):
+    import mars.dataframe as md
+    import mars.tensor as mt
+
+    # Create a mars cluster containing 2 workers
+    client = o.create_mars_cluster(2, 4, 16, min_worker_num=1)
+    result = md.DataFrame(mt.random.rand(1000, 3)).sum().execute()
+    client.stop_server()
+```
+
+Tell the worker to execute the data transformation the dependency of which is not included in PyODPS node but in function image. `data_preprocessor.py` contains two functions `a_python_func` (execute in PyODPS node) and `another_python_func` (execute in the mars worker pod).
+
+```Python
+def a_python_func(param_a, param_b):
+    import mars.remote as mr
+
+    def func():
+        import sys
+        sys.path.append('/run/my_func')
+
+        from data_preprocessor import another_python_func
+        another_python_func(param_a, param_b)
+
+    # Create a mars cluster containing 1 worker
+    client = o.create_mars_cluster(1, 4, 16)
+    # Use mars remote api to tell the worker execute func.
+    mr.spawn(func, args=()).execute()
+
+    client.stop_server()
+
+
+def another_python_func(param_a, param_b):
+    import tsfresh
+
+    # Do data transformation using the packages which
+    # are not included in PyODPS node but included in
+    # the function image.
+```
+
+#### Function Standards
+
+The parameters of the python function for `TO RUN` contain two parts:
+
+1. Context from the SQLFlow statement, it contains the following required fields.
+
+- db_type: mysql/hive/maxcompute, from sqlflowserver configuration.
+- input_table: the table name from the clause `SELECT * FROM input_table`.
+- output_table: the table name from the clause `INTO output_table`.
+- image_name: the docker image name from the clause `TO RUN image_name/func_name`.
+
+2. Attributes from `WITH` clause.
+
+Please check the following example. The first four parameters are from the context and param_1 ~ param_n are from `WITH` Attributes.
+
+```Python
+def a_python_func(
+    db_type,
```

Comment:
Let us define the first parameter being a Python dictionary named `context`.  Users use the WITH clause to specify the rest of the parameters.

Reply:
Good idea.